app/src/activity.c

```diff
@@ -126,7 +130,14 @@ void activity_work_handler(struct k_work *work) {
     int32_t current = k_uptime_get();
     int32_t inactive_time = current - activity_last_uptime;
 #if IS_ENABLED(CONFIG_ZMK_SLEEP)
-    if (inactive_time > MAX_SLEEP_MS && !is_usb_power_present()) {
+    if (inactive_time > MAX_SLEEP_MS &&
+        !is_usb_power_present()
+#if IS_ENABLED(CONFIG_ZMK_AWAKE_WHILE_BLE_CONNECTED)
+        // if user inactive and no USB
+        // keyboard will sleep as soon as BLE is disconnected
+        && !zmk_ble_active_profile_is_connected()
```

Comment:
Got this working on the peripheral side with:
```suggestion
#if !IS_ENABLED(CONFIG_ZMK_SPLIT) || IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
        && !zmk_ble_active_profile_is_connected()
#else
        && !zmk_split_bt_peripheral_is_connected()
#endif
```